initializr-generator/src/main/java/io/spring/initializr/generator/project/ProjectGenerator.java

```diff
@@ -121,11 +121,20 @@ public ProjectGenerator(Consumer<ProjectGenerationContext> contextConsumer) {
 		return () -> {
 			if (description instanceof MutableProjectDescription) {
 
-				context.registerBean(ProjectDescriptionModifications.class, () ->
-						new ProjectDescriptionModifications((MutableProjectDescription) description));
+				// NOTE: This PD is created via the request converter in the web tier so it could very well be
+				// MutableProjectDescription or CustomExtProjectDescription, or WhateverProjectDescription
+
+				// TODO in order to support custom descriptions and possibly custom diffs we need to add a factory
+				// that will create the diff and the copy/clone - otherwise its hardcoded to MPD
+
+				final MutableProjectDescription mutableProjectDesc = (MutableProjectDescription) description;
+				final MutableProjectDescription mutableProjectDescCopy = new MutableProjectDescription(mutableProjectDesc);
+				final ProjectDescriptionDiff<MutableProjectDescription> diff = new ProjectDescriptionDiff(mutableProjectDescCopy);
```

Comment:
Here is where its breaking down. Will need a diff factory if we want to support separate diff objects